include/highfive/bits/H5Reference_misc.hpp

```diff
@@ -45,7 +46,7 @@ inline T Reference::dereference(const Object& location) const {
         HDF5ErrMapper::ToException<ReferenceException>(
             "Trying to dereference the wrong type");
     }
-    return T(obj);
+    return std::move(obj);
```

Comment:
I feel like I keep going back to the same question over and over.. but why the move specifically here? I thought this is a case for copy-illision?

Reply:
In this case we can't have copy-illision, we have different classes (or am I missing something?)
What I did was to transform a constructor by coping the base class to a constructor which moves it (since this PR implements Object move ctor).

Reply:
After discussion all clear. thank you!